app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -249,13 +249,17 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
 
     override fun onDisplayPreferenceDialog(preference: Preference) {
         if (preference is SsidPreference) {
-            lateinit var permissionsToCheck: Array<String>
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                permissionsToCheck = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
-            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                permissionsToCheck = arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION)
+            val permissionsToCheck: Array<String> = when {
+                Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q -> {
+                    arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
+                }
+                Build.VERSION.SDK_INT >= Build.VERSION_CODES.O -> {
+                    arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION)
+                }
+                else -> {
+                    arrayOf()
```

Comment:
This should not be blank because we still require permissions for older versions of Android.  The permissions should match the other sensors that rely on it.  This is required for us to get SSID info.

https://github.com/home-assistant/android/blob/master/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L504-L513

Reply:
I'm not sure that that is correct.  The app works all the way back to Android 5, but the need to request runtime permissions was not introduced till Android 6.

There may well be a second bug here, where the course location check should be against M not O, but I think there are still Android versions for which an empty array is the correct answer.  It's interesting that there is a second place in the code making the same decisions.  Clearly that code should not be duplicated.

Reply:
> It's interesting that there is a second place in the code making the same decisions. Clearly that code should not be duplicated.

This is because sensors as I linked to have separate permission checks that are part of sensor architecture.  The code that you are touching was actually for App Configuration and we also decided to inform the user upon app launch that they are missing permissions or dont have GPS enabled.  The permission and location check is required for us to get certain data.  So the permission check for sensors is still required.

The code that I linked to has been there for a very long time and my own devices on Android 5 do not complain about the run time permissions.  They are simply ignored I think?

Reply:
If it all works fine on Android 5, then that sounds great.  It seems therefore you have this all in hand in the duplicate you linked - should I just close this PR?